Major update to camera driver.

RobWks opened the pull request:
**I have re-written the camera driver to include the following new features:**

1. Image resolution is automatically adjusted appropriately for the sensor.
2. Automatic adjustment of contrast and white balance with optional brightness reference.
3. Configurations can be saved in uEye Cockpit and used with APE.
4. The configuration created by the AutoConfiguration method can be saved to a file.
5. A non-blocking live feed in a separate thread can be opened and closed.
6. A non-blocking recording can be started in a separate thread and be opened and closed. 

**Tested capabilities:**

- Can connect and record video from multiple cameras at the same time.
- For each individual camera, recordings can be started and stopped multiple times.
- Multiple images can be captured while recordings and/or live feeds are running.
- A live feed can be viewed while the AutoConfiguration method is running. 
- A windows save/load dialog will popup if a path is not supplied when loading or saving configurations.

**Known limitations:**

- The live feed can only be started and stopped once. It is currently not possible to reopen the window displaying the live feed after it has been closed.*
- When using multiple cameras, only a single live feed can be run.*

*These issues arise from using Qt across multiple threads. 

I have tested this code using an IDS UI-3080CP-C-HQ R2 and an IDS UI-3250CP-C-HQ R2 connected to my personal Windows 10 computer. I attempted to test the code on a PC connected to an Aerotech system in the lab, but I ran into issues with the ueye.dll crashing. It does not seem to be related to the changes I have made to the camera driver. I have included the _camera_3250CP2.py_ and _IDS_ueye_3250CP2.py_, though you can remove them afterwards.

RobWks committed:
Initial commit of enhanced camera functionality.

RobWks committed:
Update record loop to adjust FPS for each device.

RobWks committed:
Slight increase to brightness_reference.

RobWks commented:
Once I get the .dll issue fixed in the lab, I can test this during a print. You can also do any testing you'd like using my branch.

jhardin4 commented:
There are a couple things missing from the Device:
1.  The eproc definitions in requirements
2.  A way of manually configuring the settings of the camera.

Can you get those are do you want me to?

RobWks committed:
Added in manual configuration of camera settings.

User controllable parameters included are:
1. Exposure time
2. Gain
3. Black level
4. Gamma

jhardin4 committed:
Updated requirements
